### PR TITLE
Fixes #19001 - improved file stubbing in tests

### PR DESCRIPTION
**Pull request by lzap**
Nothing more clever comes to my mind.

### CODE REVIEW

File: app/controllers/foreman_bootdisk/api/v2/disks_controller.rb [hunk 1]
```diff
@@ -32,16 +32,22 @@ def host
...
           else
             tmpl = host.bootdisk_template_render
             ForemanBootdisk::ISOGenerator.generate(:ipxe => tmpl) do |iso|
-              send_data File.read(iso), :filename => "#{host.name}.iso"
+              send_data read_file(iso), :filename => "#{host.name}.iso"
             end
           end
         end
 
+        protected
```
Comment (domcleal) [inline app/controllers/foreman_bootdisk/api/v2/disks_controller.rb]: Not sure that this visibility is necessary, `private` is sufficient.

File: app/controllers/foreman_bootdisk/disks_controller.rb [hunk 2]
```diff
@@ -53,13 +53,19 @@ def subnet
...
       ForemanBootdisk::ISOGenerator.generate(:ipxe => tmpl) do |iso|
-        send_data File.read(iso), :filename => "bootdisk_subnet_#{subnet.name}.iso"
+        send_data read_file(iso), :filename => "bootdisk_subnet_#{subnet.name}.iso"
       end
     end
 
     def help
     end
 
+    protected
```
Comment (domcleal) [inline app/controllers/foreman_bootdisk/disks_controller.rb]: Ditto.

**Comment by lzap** [comment]
Right I was unsure, thanks.

**Comment by domcleal** [comment]
Merged, thanks @lzap.

**Pull request merged**